BUILD

```diff
@@ -0,0 +1,131 @@
+repo = ["WORKSPACE",
+         ".bazelrc",
+         "//scala:sources",
+         "//scala/support:sources",
+         "//src/java/com/google/devtools/build/lib:sources",
+         "//src/java/io/bazel/rulesscala/io_utils:sources",
+         "//src/java/io/bazel/rulesscala/jar:sources",
+         "//src/java/io/bazel/rulesscala/scalac:sources",
+         "//src/java/io/bazel/rulesscala/worker:sources",
+         "//src/scala:sources",
+         "//src/scala/io/bazel/rules_scala/scrooge_support:sources",
+         "//src/scala/io/bazel/rules_scala/tut_support:sources",
+         "//src/scala/scripts:sources",
+         "//test:sources",
+         "//test/data:sources",
+         "//test/src/main/resources/scala/test:sources",
+         "//test/src/main/scala/scala/test/srcjars:sources",
+         "//test/src/main/scala/scala/test/twitter_scrooge:sources",
+         "//test/src/main/scala/scala/test/twitter_scrooge/thrift:sources",
+         "//test/src/main/scala/scala/test/twitter_scrooge/thrift/thrift2:sources",
+         "//test/src/main/scala/scala/test/twitter_scrooge/thrift/thrift2/thrift3:sources",
+         "//test/src/main/scala/scala/test/twitter_scrooge/thrift/thrift2/thrift4:sources",
+         "//test/tut:sources",
+         "//test_expect_failure/disappearing_class:sources",
+         "//test_expect_failure/scala_library_suite:sources",
+         "//test_expect_failure/transitive/java_to_scala:sources",
+         "//test_expect_failure/transitive/scala_to_java:sources",
+         "//test_expect_failure/transitive/scala_to_scala:sources",
+         "//thrift:sources",
+         "//tut_rule:sources",
+         "//twitter_scrooge:sources"]
+
+additions = [] 
+
+sh_test(
+    name = "build_targets_under_tests_package",
+    size = "large",
+    data = repo,
```

Comment:
If you want the toolchain to be shiped in the test you need to pass  "@local_jdk//:jdk", in the data dependency. You will also need bazel itself as a dependency and specify it to the --javabase option of bazel.

See https://github.com/bazelbuild/bazel/blob/master/src/test/shell/testenv.sh#L246
and https://github.com/bazelbuild/bazel/blob/master/src/test/shell/testenv.sh#L87